lib/web/apiserver.go

```diff
@@ -283,160 +289,163 @@ func NewHandler(cfg Config, opts ...HandlerOption) (*APIHandler, error) {
-	h.GET("/webapi/oidc/callback", h.WithMetaRedirect(h.oidcCallback))
-	h.POST("/webapi/oidc/login/console", httplib.MakeHandler(h.oidcLoginConsole))
-
-	// SAML 2.0 handlers
-	h.POST("/webapi/saml/acs", h.WithRedirect(h.samlACS))
-	h.GET("/webapi/saml/sso", h.WithMetaRedirect(h.samlSSO))
-	h.POST("/webapi/saml/login/console", httplib.MakeHandler(h.samlSSOConsole))
-
-	// Github connector handlers
-	h.GET("/webapi/github/login/web", h.WithRedirect(h.githubLoginWeb))
-	h.GET("/webapi/github/callback", h.WithMetaRedirect(h.githubCallback))
-	h.POST("/webapi/github/login/console", httplib.MakeHandler(h.githubLoginConsole))
-
-	// MFA public endpoints.
-	h.POST("/webapi/mfa/login/begin", h.withLimiter(challengeLimiter, h.mfaLoginBegin))
-	h.POST("/webapi/mfa/login/finish", httplib.MakeHandler(h.mfaLoginFinish))
-	h.POST("/webapi/mfa/login/finishsession", httplib.MakeHandler(h.mfaLoginFinishSession))
-	h.DELETE("/webapi/mfa/token/:token/devices/:devicename", httplib.MakeHandler(h.deleteMFADeviceWithTokenHandle))
-	h.GET("/webapi/mfa/token/:token/devices", httplib.MakeHandler(h.getMFADevicesWithTokenHandle))
-	h.POST("/webapi/mfa/token/:token/authenticatechallenge", httplib.MakeHandler(h.createAuthenticateChallengeWithTokenHandle))
-	h.POST("/webapi/mfa/token/:token/registerchallenge", httplib.MakeHandler(h.createRegisterChallengeWithTokenHandle))
-
-	// MFA private endpoints.
-	h.GET("/webapi/mfa/devices", h.WithAuth(h.getMFADevicesHandle))
-	h.POST("/webapi/mfa/authenticatechallenge", h.WithAuth(h.createAuthenticateChallengeHandle))
-	h.POST("/webapi/mfa/devices", h.WithAuth(h.addMFADeviceHandle))
-	h.POST("/webapi/mfa/authenticatechallenge/password", h.WithAuth(h.createAuthenticateChallengeWithPassword))
-
-	// trusted clusters
-	h.POST("/webapi/trustedclusters/validate", httplib.MakeHandler(h.validateTrustedCluster))
-
-	// User Status (used by client to check if user session is valid)
-	h.GET("/webapi/user/status", h.WithAuth(h.getUserStatus))
-
 	// Issue host credentials.
 	h.POST("/webapi/host/credentials", httplib.MakeHandler(h.hostCredentials))
 
-	h.GET("/webapi/roles", h.WithAuth(h.getRolesHandle))
-	h.PUT("/webapi/roles", h.WithAuth(h.upsertRoleHandle))
-	h.POST("/webapi/roles", h.WithAuth(h.upsertRoleHandle))
-	h.DELETE("/webapi/roles/:name", h.WithAuth(h.deleteRole))
+	if !cfg.MinimalReverseTunnelRoutesOnly {
 
-	h.GET("/webapi/github", h.WithAuth(h.getGithubConnectorsHandle))
-	h.PUT("/webapi/github", h.WithAuth(h.upsertGithubConnectorHandle))
-	h.POST("/webapi/github", h.WithAuth(h.upsertGithubConnectorHandle))
-	h.DELETE("/webapi/github/:name", h.WithAuth(h.deleteGithubConnector))
+		// ping endpoint is used to check if the server is up. the /webapi/ping
+		// endpoint returns the default authentication method and configuration that
+		// the server supports. the /webapi/ping/:connector endpoint can be used to
+		// query the authentication configuration for a specific connector.
+		h.GET("/webapi/ping", httplib.MakeHandler(h.ping))
```

Comment:
I see the original issue states that `/webapi/ping` is needed, but I assume this isn't actually the case.

Reply:
Right. The agent uses `/webapi/find` instead.